MetadataExtractor/Formats/QuickTime/QuickTimeMetadataReader.cs

```diff
@@ -113,6 +115,59 @@ void UuidHandler(AtomCallbackArgs a)
                 }
             }
 
+            void MetaDataHandler(AtomCallbackArgs a)
+            {
+                // see https://developer.apple.com/library/archive/documentation/QuickTime/QTFF/Metadata/Metadata.html
+                switch (a.TypeString)
+                {
+                    case "keys":
+                    {
+                        var version = a.Reader.GetByte();
+                        var flags = a.Reader.GetBytes(3);
+                        var entryCount = a.Reader.GetUInt32();
+                        for (int i = 1; i <= entryCount; i++)
+                        {
+                            var keySize = a.Reader.GetUInt32();
+                            var keyValueSize = (int)keySize - 8;
+                            var keyNamespace = a.Reader.GetUInt32();
+                            var keyValue = a.Reader.GetBytes(keyValueSize);
+                            metaDataKeys.Add(Encoding.UTF8.GetString(keyValue));
+                        }
+                        break;
+                    }
+                    case "ilst":
+                    {
+                        var directory = new QuickTimeMetadataHeaderDirectory();
+                        // Iterate over the list of Metadata Item Atoms.
+                        for (int i = 0; i < metaDataKeys.Count; i++)
+                        {
+                            long atomSize = a.Reader.GetUInt32();
+                            var atomType = a.Reader.GetUInt32();
+
+                            // Indexes into the metadata item keys atom are 1-based (1�entry_count).
+                            // atom type for each metadata item atom should be set equal to the index of the key
+                            var key = metaDataKeys[(int)atomType - 1];
```

Comment:
Please do a bounds check here and, if out of bounds, add an error to the directory rather than throwing an exception.